server/lib/hostlib.js

```diff
@@ -4,20 +4,28 @@ import _ from 'lodash';
 import { convertToCurrency } from '../lib/currency';
 import models, { Op, sequelize } from '../models';
 
-export function getHostedCollectives(hostid, endDate = new Date()) {
+export function getHostedCollectives(hostid, startDate, endDate = new Date()) {
   return sequelize.query(
     `
-    SELECT g.* FROM "Collectives" g
-    LEFT JOIN "Members" ug ON g.id = ug."CollectiveId"
-    WHERE ug.role='HOST'
-      AND ug."MemberCollectiveId"=:hostid
-      AND g."deletedAt" IS NULL
-      AND ug."deletedAt" IS NULL
-      AND ug."createdAt" < :endDate
-      AND g."createdAt" < :endDate
+    with "members" as (
+      SELECT m."CollectiveId"
+      FROM "Members" m
+      WHERE m.role='HOST'
+        AND m."MemberCollectiveId" = :hostid
+        AND (m."deletedAt" IS NULL OR m."deletedAt" > :startDate)
+        AND m."createdAt" < :endDate
+    ), "transactions" as (
+      SELECT DISTINCT t."CollectiveId" FROM "Transactions" t WHERE t."HostCollectiveId" = :hostid AND t."createdAt" < :endDate AND t."createdAt" > :startDate
+    )
```

Comment:
I had to go through transactions since the membership record was hard-deleted when the collectives were moved to the new host.